src/main/java/javaslang/collection/IndexedSeq.java

```diff
@@ -180,26 +207,53 @@ default T last() {
     Tuple2<? extends IndexedSeq<T>, ? extends IndexedSeq<T>> span(Predicate<? super T> predicate);
 
     @Override
-    default boolean startsWith(java.lang.Iterable<? extends T> that, int offset) {
-        Objects.requireNonNull(that, "that is null");
-        final java.util.Iterator<? extends T> thatIter = that.iterator();
-        if (!thatIter.hasNext()) {
-            return true;
-        }
-        final int length = length();
-        if (offset < 0 || offset > length) {
-            return false;
+    default boolean startsWith(Iterable<? extends T> that, int offset) {
```

Comment:
actually here if it's null it calls the super method that has the null check! :wink: